### PR TITLE
Proxy server

### PR DESCRIPTION
**Pull request by spectraldoy**

### CODE REVIEW

File: server.cpp [hunk 1]
```diff
@@ -20,7 +20,8 @@
 #include <sstream>
 #include <ctime>
 #include <regex>
-
+#include <iostream>
+#include <arpa/inet.h>
```
Comment (spectraldoy) [inline server.cpp]: we already have arpa/inet.h included earlier

File: server.cpp [hunk 2]
```diff
@@ -197,16 +199,20 @@ void handle_request(struct server_app *app, int client_socket) {
...
+    //std::cout<<requested_path;
     // TODO: Implement proxy and call the function under condition
     // specified in the spec
-    // if (need_proxy(...)) {
-    //    proxy_remote_file(app, client_socket, file_name);
-    // } else {
-    serve_local_file(client_socket, requested_path);
-    //}
+    if (requested_path.length() >= 3) {
+        if (requested_path.substr(requested_path.length() - 3, 3) == ".ts") {
```
Comment (spectraldoy) [inline server.cpp]: can make this one if statement using &&
Could also consider using the determine_filetype function, adding a new `ft_ts` enum

File: server.cpp [hunk 3]
```diff
@@ -303,11 +309,53 @@ void proxy_remote_file(struct server_app *app, int client_socket, const char *re
...
-    char response[] = "HTTP/1.0 501 Not Implemented\r\n\r\n";
-    send(client_socket, response, strlen(response), 0);
-}
+    if ((client_fd = socket(PF_INET, SOCK_STREAM, 0)) == 0) {
+        perror("socket failed");
+        exit(EXIT_FAILURE);
+    }
 
+    their_addr.sin_family = AF_INET; 
+    their_addr.sin_port = htons(PORT);
```
Comment (spectraldoy) [inline server.cpp]: isn't htons supposed to take a hostname as an argument? Maybe this is what you're doing, I'm not entirely sure

Comment (Ajzboss) [reply server.cpp]: htons just changes endianness for the port number, since it's an int to make it network-readable

Comment (spectraldoy) [reply server.cpp]: I see

**Pull request merged**